### PR TITLE
feat: EntityVerticle registry RFC

### PR DESCRIPTION
**Pull request by halber**

### CODE REVIEW

File: docs/rfc/001-registry-service-rfc.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+This RFC introduces a more robust approach to EntityVerticle service discovery and registration within the NeonBee ecosystem, addressing the limitations of the existing system. The proposal aims to improve reliability, scalability, and maintainability by ensuring that services are dynamically discoverable and resilient to change.
```
Comment (carlspring) [inline docs/rfc/001-registry-service-rfc.md:1]:
```suggestion
This RFC introduces a more robust approach to `EntityVerticle` service discovery and registration within the NeonBee ecosystem, addressing the limitations of the existing system. The proposal aims to improve reliability, scalability, and maintainability by ensuring that services are dynamically discoverable and resilient to change.
```

File: docs/rfc/001-registry-service-rfc.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Motivation
+
+This proposal seeks to address problems in our microservices architecture, including EntityVerticle registration failures, leading to outages and performance degradation. The primary motivations are:
```
Comment (carlspring) [inline docs/rfc/001-registry-service-rfc.md:5]:
```suggestion
This proposal seeks to address problems in our microservices architecture, including `EntityVerticle` registration failures, leading to outages and performance degradation. The primary motivations are:
```

File: docs/rfc/001-registry-service-rfc.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+- **Decoupling Implementation**: By making EntityVerticle registration configurable, we enable:
```
Comment (carlspring) [inline docs/rfc/001-registry-service-rfc.md:7]:
```suggestion
- **Decoupling Implementation**: By making `EntityVerticle` registration configurable, we enable:
```

**Review by carlspring — DISMISSED** [review]
LGTM! :)

File: docs/rfc/001-registry-service-rfc.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+This RFC introduces a more robust approach to `EntityVerticle` service discovery and registration within the NeonBee ecosystem, addressing the limitations of the existing system. The proposal aims to improve reliability, scalability, and maintainability by ensuring that services are dynamically discoverable and resilient to change.
+
+## Motivation
+
+This proposal seeks to address problems in our microservices architecture, including `EntityVerticle` registration failures, leading to outages and performance degradation. The primary motivations are:
+
+- **Decoupling Implementation**: By making `EntityVerticle` registration configurable, we enable:
+    - Reaction to diverse cluster events, such as Kubernetes events or liveness probes.
+    - Centralized EntityVerticle registration management.
```
Comment (carlspring) [inline docs/rfc/001-registry-service-rfc.md:9]:
```suggestion
    - Centralized `EntityVerticle` registration management.
```

**Review by carlspring — APPROVED** [review]
LGTM! :)

**Comment by pk-work** [comment]
Is this only for EntityVerticles? Or Will we have a general Registry<T> which will allow o register data? Will we also have a SelfCleaningRegistry?

Is this RFC a proposal to replace the withdrawn SelfCleaningRegistry commit?

**Comment by halber** [comment]
The intention of this RFC is to get the existing problems under control. The introduction of other features is not part of this RFC.

**Pull request merged**